com.wso2telco.dep.gw.smsapi/smsapigw-synapse/src/main/synapse-config/sequences/smsRetrieveOutSeq.xml

```diff
@@ -1,29 +1,44 @@
-	        
-	        var messages = payload.inboundSMSMessageList.inboundSMSMessage;
-	        for (var i in messages) {
-	            var messageId = messages[i].messageId;
-	            var registrationId = messages[i].destinationAddress;
-	            messages[i].resourceURL = resourceUrlPrefix + '/' + registrationId + '/' + requestId + '/' + messageId;
-	        }
-	
-	        payload.inboundSMSMessageList.resourceURL = resourceUrlPrefix + '/' + registrationId + '/' + requestId;
-	        mc.setPayloadJSON(payload);
-          "/>
+    <property name="gatewayResourceUrlPrefix" expression="get-property('GATEWAY_RESOURCE_URL_PREFIX')"/>
+    <property name="smsRetrieveResourceUrlPrefix" expression="get-property('SMS_RETRIEVE_GATEWAY_RESOURCE_URL_PREFIX')"/>
+    <filter source="boolean(get-property('smsRetrieveResourceUrlPrefix'))" regex="true">
+        <then>
+            <property name="resourceUrlPrefix" expression="get-property('smsRetrieveResourceUrlPrefix')"/>
+        </then>
+        <else>
+            <filter source="boolean(get-property('gatewayResourceUrlPrefix'))" regex="true">
+                <then>
+                    <property name="resourceUrlPrefix" expression="get-property('gatewayResourceUrlPrefix')"/>
+                </then>
+                <else>
+                    <!-- throw an error-->
+                </else>
+            </filter>
+        </else>
+    </filter>
+    <property name="requestId" expression="get-property('REQUEST_ID')"/>
+    <foreach id="foreach_messages" expression="//inboundSMSMessageList/inboundSMSMessage">
+        <sequence>
+            <property name="messageId" expression="//inboundSMSMessage/messageId"/>
+            <property name="registrationId" expression="//inboundSMSMessage/destinationAddress"/>
+            <property name="propertyReplacementIndex" expression="get-property('foreach_messages_FOREACH_COUNTER')" scope="default"/>
+            <property name="propertyReplacement" expression="fn:concat(syn:get-property('resourceUrlPrefix'), '/', syn:get-property('registrationId'), '/', syn:get-property('requestId'), '/', syn:get-property('messageId'))" scope="default" xmlns:fn="http://www.w3.org/2005/xpath-functions" />
+
+            <class name="com.wso2telco.dep.common.mediation.UtilMediator">
+                <property name="propertyValue" value="propertyReplacementInArray"/>
+                <property name="propertyName" value="inboundSMSMessage"/>
+                <property name="propertyPath" value="payload.inboundSMSMessageList.inboundSMSMessage"/>
+                <property name="msgContextProperty" value="propertyReplacement"/>
+            </class>
+        </sequence>
+    </foreach>
+
+    <property name="propertyReplacement" expression="fn:concat(syn:get-property('resourceUrlPrefix'), '/', syn:get-property('registrationId'), '/', syn:get-property('requestId'))" scope="default" xmlns:fn="http://www.w3.org/2005/xpath-functions" />
```

Comment:
Use meaningful property name instead of using "propertyReplacement".  Please change it in other sequences too.

Reply:
resolved